### PR TITLE
Ensure we deserialize properly, even in the presence of values we don't understand

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs [hunk 1]
```diff
@@ -386,13 +389,12 @@ static bool TryDeserializeOptionKey(ObjectReader reader, ILookup<string, IOption
...
                     {
-                        deserializedOptionKey = new OptionKey(option, language);
-                        return true;
+                        return new OptionKey(option, language);
                     }
                 }
 
-                deserializedOptionKey = default;
-                return false;
+                Debug.Fail($"Failed to deserialize: {name}-{feature}-{isPerLanguage}-{language}");
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs:396]: uncertain if we should be doing this.  is this normal that we aren't able to deserialize an option?  or is this exceptional.

File: src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs [hunk 2]
```diff
@@ -362,7 +365,7 @@ public static SerializableOptionSet Deserialize(ObjectReader reader, IOptionServ
             var changedKeysBuilder = ImmutableHashSet.CreateBuilder<OptionKey>();
             for (var i = 0; i < count; i++)
             {
-                if (TryDeserializeOptionKey(reader, lookup, out var optionKey))
+                if (TryDeserializeOptionKey(reader, lookup) is OptionKey optionKey)
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs:368]:
```suggestion
                if (TryDeserializeOptionKey(reader, lookup) is { } optionKey)
```

File: src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs (hunk 1 above)
Comment (sharwell) [reply src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs:396]: Not sure. Is there a way to find out what option failed for the previous case?

**Review by sharwell — APPROVED** [review]

File: src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs (hunk 1 above)
Comment (genlu) [reply src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs:396]: I can try to debug after applying this fix to find out

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs:396]: thanks.  that works for me.

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs:396]:
```suggestion
```

**Pull request merged**